can't build with musl

Hello. fast_io from git master

```cpp
#include <fast_io.h>

int main () {}
```

```bash
$ gcc -dumpmachine
x86_64-alpine-linux-musl
$ g++ -std=c++20 -Ibuild/_deps/fastio-src/include main.cpp
```

fails to build on alpine linux with the following message:
```
In file included from build/_deps/fastio-src/include/fast_io_hosted/platforms/native_base.h:23,
                 from build/_deps/fastio-src/include/fast_io_hosted/platforms/native.h:2,
                 from build/_deps/fastio-src/include/fast_io_hosted.h:69,
                 from build/_deps/fastio-src/include/fast_io.h:9,
                 from main.cpp:1:
build/_deps/fastio-src/include/fast_io_hosted/platforms/posix.h: In function 'fast_io::posix_file_status fast_io::details::fstat_impl(int)':
build/_deps/fastio-src/include/fast_io_hosted/platforms/posix.h:889:23: error: aggregate 'fast_io::details::fstat_impl(int)::stat64 st' has incomplete type and cannot be defined
  889 |         struct stat64 st;
      |                       ^~
build/_deps/fastio-src/include/fast_io_hosted/platforms/posix.h:901:1: error: 'fstat64' was not declared in this scope; did you mean 'stat64'?
  901 | fstat64
      | ^~~~~~~
      | stat64
In file included from build/_deps/fastio-src/include/fast_io_hosted/file_loaders/posix_file_loader.h:3,
                 from build/_deps/fastio-src/include/fast_io_hosted/file_loaders/impl.h:3,
                 from build/_deps/fastio-src/include/fast_io_hosted.h:70:
/usr/include/linux/stat.h: At global scope:
/usr/include/linux/stat.h:56:8: error: redefinition of 'struct statx_timestamp'
   56 | struct statx_timestamp {
      |        ^~~~~~~~~~~~~~~
In file included from build/_deps/fastio-src/include/fast_io_hosted/platforms/posix.h:16:
/usr/include/sys/stat.h:124:8: note: previous definition of 'struct statx_timestamp'
  124 | struct statx_timestamp {
      |        ^~~~~~~~~~~~~~~
/usr/include/linux/stat.h:99:8: error: redefinition of 'struct statx'
   99 | struct statx {
      |        ^~~~~
/usr/include/sys/stat.h:129:8: note: previous definition of 'struct statx'
  129 | struct statx {
      |        ^~~~~
In file included from build/_deps/fastio-src/include/fast_io_hosted/filesystem/native.h:8,
                 from build/_deps/fastio-src/include/fast_io_hosted.h:84:
build/_deps/fastio-src/include/fast_io_hosted/filesystem/posix_at.h: In function 'fast_io::posix_file_status fast_io::details::posix_fstatat_impl(int, const char*, int)':
build/_deps/fastio-src/include/fast_io_hosted/filesystem/posix_at.h:145:23: error: aggregate 'fast_io::details::posix_fstatat_impl(int, const char*, int)::stat64 buf' has incomplete type and cannot be defined
  145 |         struct stat64 buf;
      |                       ^~~
```
The same command works just fine in archlinux container.
```bash
$ gcc -dumpmachine
x86_64-pc-linux-gnu
```

--- a/include/fast_io_hosted/file_loaders/file_size.h
+++ b/include/fast_io_hosted/file_loaders/file_size.h
@@ -44,7 +44,7 @@ inline ::std::size_t win32_load_file_get_file_size(void *handle)
 
 namespace fast_io::details
 {
-struct statx_timestamp {
+struct linux_statx_timestamp {
 ::std::int_least64_t tv_sec;
 ::std::uint_least32_t tv_nsec, pad;
 };
@@ -62,10 +62,10 @@ struct linux_struct_statx {
 ::std::uint_least64_t stx_size;
 ::std::uint_least64_t stx_blocks;
 ::std::uint_least64_t stx_attributes_mask;
-linux_statx_timestamp stx_atime;
-linux_statx_timestamp stx_btime;
-linux_statx_timestamp stx_ctime;
-linux_statx_timestamp stx_mtime;
+::fast_io::details::linux_statx_timestamp stx_atime;
+::fast_io::details::linux_statx_timestamp stx_btime;
+::fast_io::details::linux_statx_timestamp stx_ctime;
+::fast_io::details::linux_statx_timestamp stx_mtime;
 ::std::uint_least32_t stx_rdev_major;
 ::std::uint_least32_t stx_rdev_minor;
 ::std::uint_least32_t stx_dev_major;

--- a/include/fast_io_hosted/file_loaders/file_size.h
+++ b/include/fast_io_hosted/file_loaders/file_size.h
@@ -1,9 +1,5 @@
 #pragma once
 
-#if defined(__linux__) && defined(__NR_statx) && !defined(__statx_defined)
-#include <linux/stat.h>
-#endif
-
 #if defined(__has_builtin)
 #if !__has_builtin(__builtin_malloc) || !__has_builtin(__builtin_free)
 #include <cstdlib>
@@ -49,6 +45,32 @@ inline ::std::size_t win32_load_file_get_file_size(void *handle)
 namespace fast_io::details
 {
 
+struct linux_struct_statx {
+uint32_t stx_mask;
+uint32_t stx_blksize;
+uint64_t stx_attributes;
+uint32_t stx_nlink;
+uint32_t stx_uid;
+uint32_t stx_gid;
+uint16_t stx_mode;
+uint16_t pad0[1];
+uint64_t stx_ino;
+uint64_t stx_size;
+uint64_t stx_blocks;
+uint64_t stx_attributes_mask;
+struct statx_timestamp stx_atime;
+struct statx_timestamp stx_btime;
+struct statx_timestamp stx_ctime;
+struct statx_timestamp stx_mtime;
+uint32_t stx_rdev_major;
+uint32_t stx_rdev_minor;
+uint32_t stx_dev_major;
+uint32_t stx_dev_minor;
+uint64_t pad1[14];
+};
+
+inline constexpr ::std::uint_least32_t linux_statx_size{0x200U};
+
 inline ::std::size_t posix_loader_get_file_size(int fd)
 {
 #if defined(_WIN32) && !defined(__BIONIC__) && !defined(__WINE__)
@@ -58,7 +80,7 @@ inline ::std::size_t posix_loader_get_file_size(int fd)
 #elif defined(__linux__) && defined(__NR_statx)
 	// Linux kernel provides new __NR_statx syscall. That allows us to only extract part of information and avoid libc
 	// braindeath beneath.
-	struct statx statxbuf;
+	linux_struct_statx statxbuf;
 	system_call_throw_error(system_call<__NR_statx, int>(fd, reinterpret_cast<char const *>(u8""),
 #if defined(AT_EMPTY_PATH)
 														 AT_EMPTY_PATH
@@ -67,7 +89,7 @@ inline ::std::size_t posix_loader_get_file_size(int fd)
 																// might not be available for all libcs
 #endif
 														 ,
-														 STATX_SIZE, __builtin_addressof(statxbuf)));
+														 linux_statx_size, __builtin_addressof(statxbuf)));
 	using stx_size_unsigned_type = ::std::make_unsigned_t<decltype(statxbuf.stx_size)>;
 	if constexpr (sizeof(stx_size_unsigned_type) > sizeof(::std::size_t))
 	{

--- a/include/fast_io_hosted/file_loaders/file_size.h
+++ b/include/fast_io_hosted/file_loaders/file_size.h
@@ -44,29 +44,33 @@ inline ::std::size_t win32_load_file_get_file_size(void *handle)
 
 namespace fast_io::details
 {
+struct statx_timestamp {
+::std::int_least64_t tv_sec;
+::std::uint_least32_t tv_nsec, pad;
+};
 
 struct linux_struct_statx {
-uint32_t stx_mask;
-uint32_t stx_blksize;
-uint64_t stx_attributes;
-uint32_t stx_nlink;
-uint32_t stx_uid;
-uint32_t stx_gid;
-uint16_t stx_mode;
-uint16_t pad0[1];
-uint64_t stx_ino;
-uint64_t stx_size;
-uint64_t stx_blocks;
-uint64_t stx_attributes_mask;
-struct statx_timestamp stx_atime;
-struct statx_timestamp stx_btime;
-struct statx_timestamp stx_ctime;
-struct statx_timestamp stx_mtime;
-uint32_t stx_rdev_major;
-uint32_t stx_rdev_minor;
-uint32_t stx_dev_major;
-uint32_t stx_dev_minor;
-uint64_t pad1[14];
+::std::uint_least32_t stx_mask;
+::std::uint_least32_t stx_blksize;
+::std::uint_least64_t stx_attributes;
+::std::uint_least32_t stx_nlink;
+::std::uint_least32_t stx_uid;
+::std::uint_least32_t stx_gid;
+::std::uint_least16_t stx_mode;
+::std::uint_least16_t pad0[1];
+::std::uint_least64_t stx_ino;
+::std::uint_least64_t stx_size;
+::std::uint_least64_t stx_blocks;
+::std::uint_least64_t stx_attributes_mask;
+linux_statx_timestamp stx_atime;
+linux_statx_timestamp stx_btime;
+linux_statx_timestamp stx_ctime;
+linux_statx_timestamp stx_mtime;
+::std::uint_least32_t stx_rdev_major;
+::std::uint_least32_t stx_rdev_minor;
+::std::uint_least32_t stx_dev_major;
+::std::uint_least32_t stx_dev_minor;
+::std::uint_least64_t pad1[14];
 };
 
 inline constexpr ::std::uint_least32_t linux_statx_size{0x200U};

--- a/include/fast_io_hosted/filesystem/posix_at.h
+++ b/include/fast_io_hosted/filesystem/posix_at.h
@@ -146,7 +146,7 @@ inline posix_file_status posix_fstatat_impl(int dirfd, char const *pathname, int
 {
 #if defined(__linux__)
 
-#if !defined(__MLIBC_O_CLOEXEC) && (defined(__NR_newfstatat) || defined(__NR_fstatat64))
+#if defined(__USE_LARGEFILE64) && (defined(__NR_newfstatat) || defined(__NR_fstatat64))
 	struct stat64 buf;
 #else
 	struct stat buf;

--- a/include/fast_io_hosted/platforms/posix.h
+++ b/include/fast_io_hosted/platforms/posix.h
@@ -703,7 +703,7 @@ inline posix_file_status fstat_impl(int fd)
 {
 #if (defined(_WIN32) && !defined(__WINE__) && !defined(__BIONIC__)) && !defined(__CYGWIN__)
 	struct __stat64 st;
-#elif defined(__linux__) && !defined(__MLIBC_O_CLOEXEC)
+#elif defined(__linux__) && defined(__USE_LARGEFILE64)
 	struct stat64 st;
 #else
 	struct stat st;
@@ -715,7 +715,7 @@ inline posix_file_status fstat_impl(int fd)
 #else
 		_fstati64
 #endif
-#elif defined(__linux__) && !defined(__MLIBC_O_CLOEXEC)
+#elif defined(__linux__) && defined(__USE_LARGEFILE64)
 		fstat64
 #else
 		fstat

